apps/framework-cli/src/framework/typescript/ts_scripts/flow.ts

```diff
@@ -159,10 +156,26 @@ const startConsumer = async (
 
   await consumer.subscribe({ topics: [sourceTopic], fromBeginning: false });
   await consumer.run({
-    eachMessage: async ({ topic, partition, message }) => {
-      await handleMessage(targetTopic, flowFunction, message);
+    eachBatchAutoResolve: true,
+    eachBatch: async ({ batch }) => {
+      const messages = await Promise.all(
+        batch.messages.map((message) => handleMessage(flowFunction, message)),
+      );
+
+      const filteredMessages = messages.filter((msg) => msg !== null);
+
+      if (filteredMessages.length > 0) {
+        await producer.send({
+          topic: targetTopic,
+          messages: filteredMessages as { value: string }[],
```

Comment:
This is what I was doing wrong last time I tried batch. I was receiving a batch, but still sending one at a time. There's a noticeable difference now that both sides are batched.